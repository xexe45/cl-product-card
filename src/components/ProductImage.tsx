import { useContext } from "react";
import { ProductContext } from "./ProductCard";
import noImage from "../assets/no-image.jpg";
import styles from "../styles/styles.module.css";
import React from "react";
export interface PropsImg {
  img?: string;
  className?: string;
  style?: React.CSSProperties;
}

export const ProductImage = ({img, className, style}: PropsImg) => {

  const { product } = useContext(ProductContext);
  let imgToShow: string;
  if(img){
    imgToShow = img
  } else if(product.img) {
    imgToShow = product.img
  } else {
    imgToShow = noImage;
  }
  return (
    <img
      style={style}
      src={imgToShow}
      className={`${ styles.productImg } ${className} `}
      alt="Product"
    />
  );
};
