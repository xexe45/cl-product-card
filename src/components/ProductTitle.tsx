import { useContext } from "react";
import { ProductContext } from "./ProductCard";
import styles from "../styles/styles.module.css";
import React from "react";
export interface PropsTitle {
  title?: string;
  className?: string
  style?: React.CSSProperties;
}

export const ProductTitle = ({title = '', className, style}: PropsTitle) => {
  const { product } = useContext(ProductContext);
  const showTitle = title ? title : product.title;
  return <span style={style} className={`${ styles.productDescription } ${className} `}>{ showTitle }</span>;
};