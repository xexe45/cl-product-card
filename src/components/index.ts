import { ProductCardHOCProps } from '../interfaces/interfaces';
import { ProductButtons } from './ProductButton';
import { ProductCard as ProductCardHOC } from './ProductCard';
import { ProductImage } from './ProductImage';
import { ProductTitle } from './ProductTitle';

export * from './ProductButton';
export * from './ProductImage';
export * from './ProductTitle';
//export * from './ProductCard';

export const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
    Title: ProductTitle,
    Image: ProductImage,
    Buttons: ProductButtons
})

export default ProductCard;