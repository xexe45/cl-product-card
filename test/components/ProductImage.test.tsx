
import React from "react";
import  renderer from 'react-test-renderer'
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from "../data/products";

describe('ProductImage', () => {
    test('debe de mostrar el componente correctamente con el titulo personalizado', ()=> {
        const wrapper = renderer.create(
            <ProductImage img="https://hola.jpg" className="custom-class" />
        )
        console.log(wrapper.toJSON());

        expect(wrapper.toJSON()).toMatchSnapshot()
    });

    test('debe de mostrar el componente con el nombre del producto', ()=> {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (

                        <ProductImage />
                    )
                        
                    
                }
            </ProductCard>
        )
        console.log(wrapper.toJSON());

        expect(wrapper.toJSON()).toMatchSnapshot()
    })
})