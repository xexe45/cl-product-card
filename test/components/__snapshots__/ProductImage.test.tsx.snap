// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostrar el componente con el nombre del producto 1`] = `null`;

exports[`ProductImage debe de mostrar el componente correctamente con el titulo personalizado 1`] = `null`;
